#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/KukaBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/Youbot.proto"

WorldInfo {
  info [
    "Simulation of the KUKA youBot robot."
    "This model includes an accurate and fast simulation of the Meccanum wheels."
    "The robot can be equiped with various sensors (including the Microsoft Kinect) and configured with zero, one or two arms."
  ]
  title "KUKA youBot"
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation -0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
    ContactProperties {
      material1 "ExteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation 0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
  ]
}
Viewpoint {
  orientation 0.3609146302279128 0.06222696072402603 -0.9305205183371827 2.823433227343645
  position 11.02017043215856 7.615291082965437 11.427005678364385
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Transform {
  translation -2.45 0 0.25
  scale 1 245 4
  children [
    Shape {
      appearance Appearance {
        material Material {
          emissiveColor 0.737255 0.768627 0.737255
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -1.01 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 1
          emissiveColor 0 0.666667 1
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -0.56 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
          emissiveColor 1 0 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -0.85 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 0
          emissiveColor 1 1 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -0.7 0.01 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.333333 1 0
          emissiveColor 0.333333 1 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -3.67 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.333333 1 0
          emissiveColor 0.333333 1 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -3.5 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.666667 1
          emissiveColor 0 0.666667 1
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -4 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
          emissiveColor 1 0 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
Transform {
  translation -3.82 0.02 0
  scale 1 1 0.01
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 0
          emissiveColor 1 1 0
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
}
KukaBox {
  translation 2.4926402079193086 -1.6729168348576855 0.02493378251000437
  rotation 4.461384182445485e-12 -2.7002734124969963e-12 -1 4.962991484461816e-06
  color 1 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 -1.7929 0.02493378251000437
  rotation 1.030429020810351e-11 -2.5760590485292273e-12 -0.9999999999999999 2.451770000279148e-06
  name "Kuka box(1)"
  color 1 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 -1.9429 0.02493378251000437
  rotation -2.318462533063586e-11 -1.674449478534326e-11 -1 2.4517700003485862e-06
  name "Kuka box(2)"
  color 1 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 -2.0729 0.024933782510004374
  rotation -1.0304276048988965e-11 -9.016263687494633e-12 -1 2.4517699914911724e-06
  name "Kuka box(3)"
  color 1 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 -2.2329 0.024933782510004367
  rotation 9.01624482289139e-12 5.152154569820261e-12 -1 2.4517699959156723e-06
  name "Kuka box(4)"
  color 1 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 -2.3829 0.024933782510004374
  rotation -1.0304276048988965e-11 -9.016263687494633e-12 -1 2.4517699914911724e-06
  name "Kuka box(5)"
  color 1 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 1.7171 0.024933782510004367
  rotation 1.2880169258818447e-12 1.5456432063218922e-11 -1 2.451770000357239e-06
  name "Kuka box(6)"
  color 0 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 1.8871 0.02493378251000437
  rotation -2.318462533063586e-11 -1.674449478534326e-11 -1 2.4517700003485862e-06
  name "Kuka box(7)"
  color 0 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 2.0371 0.02493378251000437
  rotation -1.1592291962734979e-11 -1.288035615758745e-11 -1 2.451773193330269e-06
  name "Kuka box(8)"
  color 0 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 2.1771 0.024933782510004374
  rotation -1.0304276048988965e-11 -9.016263687494633e-12 -1 2.4517699914911724e-06
  name "Kuka box(9)"
  color 0 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 2.3271 0.024933782510004374
  rotation -1.0304276048988965e-11 -9.016263687494633e-12 -1 2.4517699914911724e-06
  name "Kuka box(10)"
  color 0 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 2.49263 2.5071 0.024933782510004374
  rotation -1.0304276048988965e-11 -9.016263687494633e-12 -1 2.4517699914911724e-06
  name "Kuka box(11)"
  color 0 1 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 4.40263 1.7371 0.024933782510004367
  rotation 2.060858357692049e-11 -7.728190091160424e-12 -0.9999999999999999 2.4517700003901052e-06
  name "Kuka box(12)"
  color 0 0.666667 1
  size 0.05 0.05 0.05
}
KukaBox {
  translation 4.40263 1.8971 0.02493378251000437
  rotation 3.864107578154324e-12 4.7568186169593106e-18 -1 2.4517700003653516e-06
  name "Kuka box(13)"
  color 0 0.666667 1
  size 0.05 0.05 0.05
}
KukaBox {
  translation 4.40263 2.0871 0.02493378251000437
  rotation -1.8032451768930203e-11 -4.1217170128726114e-11 -1 2.45176999569451e-06
  name "Kuka box(14)"
  color 0 0.666667 1
  size 0.05 0.05 0.05
}
KukaBox {
  translation 4.40263 2.2471 0.02493378251000437
  rotation -1.8032451768930203e-11 -4.1217170128726114e-11 -1 2.45176999569451e-06
  name "Kuka box(15)"
  color 0 0.666667 1
  size 0.05 0.05 0.05
}
KukaBox {
  translation 4.40263 2.4471 0.02493378251000437
  rotation 7.728215354735437e-12 9.418352785095367e-18 -1 2.451769995919842e-06
  name "Kuka box(16)"
  color 0 0.666667 1
  size 0.05 0.05 0.05
}
KukaBox {
  translation 4.40263 2.6371 0.02493378251000437
  rotation 7.728215354735437e-12 9.418352785095367e-18 -1 2.451769995919842e-06
  name "Kuka box(17)"
  color 0 0.666667 1
  size 0.05 0.05 0.05
}
KukaBox {
  translation 6.752564701819343 0.03709599245227633 0.02493378251000437
  rotation 3.1657824374038028e-12 -7.005815662040705e-12 -1 2.256257728007457e-06
  name "Kuka box(18)"
  color 1 0 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 6.87263 0.0370959 0.024933782510004367
  rotation -3.735307192790914e-11 2.5760671626052997e-11 -1 2.4517700005156614e-06
  name "Kuka box(19)"
  color 1 0 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 7.02263 0.0370958 0.024933782510004367
  rotation -3.735307192790914e-11 2.5760671626052997e-11 -1 2.4517700005156614e-06
  name "Kuka box(20)"
  color 1 0 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 7.15263 0.037096 0.024933782510004374
  rotation 1.030429968614184e-11 -1.030427428930536e-11 -1 2.4517700000404746e-06
  name "Kuka box(21)"
  color 1 0 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 6.652630000394794 0.03709599979026158 0.024933782510004367
  rotation 5.1115229113410736e-12 5.51696483430088e-12 -1 2.4445631508704176e-06
  name "Kuka box(22)"
  color 1 0 0
  size 0.05 0.05 0.05
}
KukaBox {
  translation 6.56269187271238 0.03709600004291994 0.024933782510004367
  rotation 1.0997297420989102e-11 -1.0511586154117236e-11 -0.9999999999999999 2.6342501972788636e-06
  name "Kuka box(23)"
  color 1 0 0
  size 0.05 0.05 0.05
}
RectangleArena {
  floorSize 25 25
  floorAppearance Appearance {
    texture ImageTexture {
      url [
        "../../../../Desktop/youbot board1.png"
      ]
    }
    textureTransform TextureTransform {
      scale 0.02 0.02
    }
  }
}
Youbot {
  translation -1.6760944602163201 0.0025691131068848383 0.10193771746818422
  rotation -0.0018594799062680483 0.042132763030163604 0.9991102905154777 0.050044508048776075
  controller "kuka_controller"
  bodySlot [
    DistanceSensor {
      translation 0.29 -5.5743e-05 -0.07
      rotation 1 0 0 1.57
      name "d1"
      lookupTable [
        0 0 0
        0.12 100000 0
      ]
      type "infra-red"
    }
    Camera {
      translation 0.26 0 0.1
      rotation 1.479159999969019e-07 -0.999999999979055 6.470559999864474e-06 -0.5236053071795865
      name "color_camera"
      fieldOfView 0.1
      width 1
      height 1
    }
    DistanceSensor {
      translation 0.27 -0.0015 -0.088
      rotation 0 1 0 1.57
      name "s0"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 -0.0085 -0.088
      rotation 0 1 0 1.57
      name "s1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 -0.0065 -0.088
      rotation 0 1 0 1.57
      name "s2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 -0.002 -0.088
      rotation 0 1 0 1.57
      name "s3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 0.0045 -0.088
      rotation 0 1 0 1.57
      name "s4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 0.0065 -0.088
      rotation 0 1 0 1.57
      name "s5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 0.0085 -0.088
      rotation 0 1 0 1.57
      name "s6"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.27 0.0015 -0.088
      rotation 0 1 0 1.57
      name "s7"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.05 0.111 -0.088176
      rotation 0 1 0 1.57
      name "left_distance_sensor"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.05 -0.151 -0.088
      rotation 0 1 0 1.57
      name "right_distance_sensor"
      type "infra-red"
    }
  ]
}
